### PR TITLE
Support laravel 5.3

### PR DESCRIPTION
**Pull request by pi0**
New Laravel queries always return collection. this is a backward compatible enhancement as Collections can be used as Arrays!

### CODE REVIEW

**Comment by lukepolo** [comment]
Hopes this gets merged, there is 1 test failing though , but is failing on master anyways

File: src/Jenssegers/Mongodb/Query/Builder.php [hunk 1]
```diff
@@ -79,6 +79,13 @@ class Builder extends BaseBuilder
     ];
 
     /**
+     * Check if we need to return Collections instead of plain arrays (laravel >= 5.3 )
+     *
+     * @var boolean
+     */
+    protected $use_collection;
```
Comment (jenssegers) [inline src/Jenssegers/Mongodb/Query/Builder.php]: This should be camel case.

File: src/Jenssegers/Mongodb/Query/Builder.php [hunk 2]
```diff
@@ -89,6 +96,8 @@ public function __construct(Connection $connection, Processor $processor)
         $this->grammar = new Grammar;
         $this->connection = $connection;
         $this->processor = $processor;
+        $s = explode('.', \Illuminate\Foundation\Application::VERSION);
+        $this->use_collection = (10 * $s[0] + $s[1]) >= 53;
```
Comment (jenssegers) [inline src/Jenssegers/Mongodb/Query/Builder.php]: Maybe this is cleaner:

```
$this->useCollection = \Illuminate\Foundation\Application::VERSION >= '5.3.0';

**Comment by alsemany** [comment]
Hello , 
I need help with relations 
some relations like morphedByMany is not working

**Comment by pi0** [comment]
@alsemany True, but not related to this PR. according to official README, Supported relations are :

- hasOne
- hasMany
- belongsTo
- belongsToMany
- embedsOne
- embedsMany

**Comment by bytesarelife** [comment]
Any idea when this will be merged? I started a project in 5.3 just to find out that this repo is not compatible yet :/

**Comment by amiri27** [comment]
@jenssegers any chance to merge this PR?

**Comment by jenssegers** [comment]
I'll merge this, but I can only tag a new release once we have fixed some of the tests though.

**Pull request merged**